IPNI_HTTP_PROVIDER.md

```diff
@@ -0,0 +1,310 @@
+
+As a special case, the following content types are interpreted as their IPLD equivelant:
+
+* `application/json` is considered to be the same as `application/vnd.ipld.dag-json`.
+* `application/cbor` is considered to be the same as `application/vnd.ipld.dag-cbor`.
+
+The common data types, such as `CID` and `Multihash`, are encoded in their
+standard [multibase format](https://github.com/multiformats/multibase) when referenced in URLs.
+
+### Versioning
+
+The IPNI Provider HTTP API adopts a versioning strategy that utilizes the path prefix to indicate major releases. As a
+best practice, future iterations that introduce breaking changes should increment the version number in the API path.
+This approach ensures compatibility and allows clients to adapt to new versions while maintaining backward compatibility
+with existing implementations. By explicitly incorporating versioning in the API path, it becomes easier to manage and
+track changes over time, facilitating smooth transitions and providing a clear indication of API evolution.
+
+
+### Namespacing
+
+This specification introduces the `/ipni/` URL prefix to namespace the functionality associated with IPNI HTTP providers. The presence of this namespace allows IPNI specifications to evolve without conflicting with other URL patterns that an HTTP server may support. Therefore, this specification reserves the right for any sub-tree that appears after this prefix.
+
+#### Implicit Namespace Assumption
+
+Indexer nodes implicitly assume the `/ipni/` URL prefix. When receiving an announcement from providers, an indexer will attempt to fetch advertisements by joining the announced URL with the paths documented in this specification. Implementers have the freedom to include additional paths to the announced URL if desired.
+
+For example, if the announced URL is `https://ipni-provider.example`, it will be accessed at `https://ipni-provider.example/ipni/v1/ad/head` to fetch the head advertisement. Similarly, if the announced URL is `https://ipni-provider.example/my/prefix`, it will be accessed at `https://ipni-provider.example/my/prefix/ipni/v1/ad/head`.
+
+#### Custom Paths in Multiaddr
+
+IPNI implementers have the option to announce HTTP publisher URLs in `Multiaddr` format. However, this format does not officially support paths. To accommodate custom paths in the announced URLs, implementers can utilize the `httpath` parameter. The handling of `httpath` is similar to that of vanilla HTTP URLs with an additional specified path.
+
+For example, if the announced multiaddr is `/dns/ipni-provider.example/https`, it will be accessed at `https://ipni-provider.example/ipni/v1/ad/head` to fetch the head advertisement. On the other hand, if the announced multiaddr is `/dns/ipni-provider.example/https/httpath/%2Fmy%2Fprefix`, it will be accessed at `https://ipni-provider.example/my/prefix/ipni/v1/ad/head`.
```

Comment:
I've left a comment here: https://github.com/ipni/go-libipni/issues/42#issuecomment-1569330073. More context there, but I don't think we should be putting application logic in the multiaddr.

Reply:
"application" is a loose term. From my perspective this is not an application logic in the same way that in a web 2 world URL to a REST API is not an application logic. It's a standard configuration for HTTP API interaction.

Reply:
Removing support for `httpath` in multiaddr as per Slack discussion.